docs/configuration.md

```diff
@@ -243,6 +243,47 @@ values, but here they are if you need to:
 ### regex
 This is the regex which is used to match the current branch to the correct branch configuration.
 
+### source-branches
+Because git is a directed graph GitVersion sometimes cannot tell which branch the current branch was branched from.
```

Comment:
I think it's valuable to point out that sometimes GitVersion can't figure out the origin of the current branch. Whether that's because it's a DAG or not is perhaps not that important.